dotCMS/src/integration-test/java/com/dotmarketing/quartz/DotStatefulJobTest.java

```diff
@@ -0,0 +1,165 @@
+        final String jobGroupName = DotStatefulJob.getJobGroupName(MyStatefulJob.class);
+        QuartzUtils.removeJob(jobName, jobGroupName);
+    }
+
+    /**
+     * Given scenario: We prep to trigger a bunch of tasks that must be executed sequentially (one at the time) never in parallel.
+     * Expected Results: first all triggered tasks must be executed and completed. second their time mark must revel that none of theme overlapped.
+     *
+     * @throws SchedulerException
+     * @throws ParseException
+     * @throws ClassNotFoundException
+     * @throws InterruptedException
+     */
+    @Test
+    public void Test_Launch_Stateful_Jobs_Verify_They_Dont_Overlap_In_Time()
+            throws SchedulerException, ParseException, ClassNotFoundException, InterruptedException {
+
+        //in order to avoid conflicts clean any reference that could have been left.
+        removeAnyExistingJob();
+
+        //Now enqueue a few jos .. They will sleep randomly to simulate work.
+        for (int i = 1; i <= MyStatefulJob.MAX_THREADS; i++) {
+            MyStatefulJob.fireJob(ImmutableMap.of("index", i));
+            //Now verify the detail has been added for the present trigger we're introducing
+            final Optional<Map<String, Object>> triggerJobDetail = DotStatefulJob
+                    .getTriggerJobDetail(MyStatefulJob.class);
+            assertNotNull(triggerJobDetail);
+            assertTrue(triggerJobDetail.isPresent());
+            final Map<String, Object> detail = triggerJobDetail.get();
+            assertNotNull(detail);
+            //The size of the detail should have grown as we insert new triggers.
+            assertEquals(detail.size(), i);
+        }
+
+        //Verify at least 1 job has been launched.
+        final Optional<JobExecutionContext> jobExecutionContext = getJobExecutionContext();
+        assertTrue(jobExecutionContext.isPresent());
+
+        //Now lets wait for the all the threads to complete.
+        MyStatefulJob.countDownLatch.await();
+
+        //Now lets revise the execution times. Since they were supposed to run sequentially the should never overlap.
+        final HashSet<MyStatefulJob> myStatefulJobs = new HashSet<>(MyStatefulJob.finishedJobs);
+        final Iterator<MyStatefulJob> iterator = myStatefulJobs.iterator();
+        while(iterator.hasNext()){
+            final MyStatefulJob myStatefulJob = iterator.next();
+            iterator.forEachRemaining(job -> {
+                //They should never overlap.
+                assertFalse(myStatefulJob.getTimeRange().overlaps(job.getTimeRange()));
+            });
+        }
+    }
+
+    /**
+     * This gets you the execution context for the given job-name.
+     * @return
+     */
+    private Optional<JobExecutionContext> getJobExecutionContext(){
+        final String jobName = DotStatefulJob.getJobName(MyStatefulJob.class);
+        try {
+            final Scheduler sequentialScheduler = QuartzUtils.getSequentialScheduler();
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'QuartzUtils.getSequentialScheduler' due to existing static import 'com.dotmarketing.quartz.QuartzUtils.getSequentialScheduler'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077927)